### PR TITLE
Homework 03. Task 1 - 6

### PR DESCRIPTION
**Pull request by B0RlS**
Probably I will add with checks / exceptions

### CODE REVIEW

File: HW_03/Boris_Tsarikov/boris_tsarikov_hw_03_array_to_hash_to_array.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Task: array to hash to array
+
+Input:
+arr = 1, 3, 6, 8
+
+script line:
+Hash[(0...arr.size).zip arr].values.flatten
+
```
Comment (labs-ruby) [inline HW_03/Boris_Tsarikov/boris_tsarikov_hw_03_array_to_hash_to_array.txt]: my irb return: Hash[(0...arr.size).zip arr].values.flatten 
=> [1, 3, 6, 8]

Comment (labs-ruby) [reply HW_03/Boris_Tsarikov/boris_tsarikov_hw_03_array_to_hash_to_array.txt]: don't understand, cold you please explain why are we need method doing nothing?

File: HW_03/Boris_Tsarikov/boris_tsarikov_hw_03_t_01.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+def task_1(str)
+  result = ''
+  str.each_line do |i|
```
Comment (labs-ruby) [inline HW_03/Boris_Tsarikov/boris_tsarikov_hw_03_t_01.rb]: line instead i - more understandable

File: HW_03/Boris_Tsarikov/boris_tsarikov_hw_03_t_03.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# There was a problem with getting the number working with the DateTime class,
+# using the help of github.com/yaourtv in the get_theme method.
+# Figured it all out now
+require 'date'
+
+begin
+  def task_3(str)
+    result = 0
+    duretion = []
```
Comment (labs-ruby) [inline HW_03/Boris_Tsarikov/boris_tsarikov_hw_03_t_03.rb]: duration

File: HW_03/Boris_Tsarikov/boris_tsarikov_hw_03_t_03.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    index = 0
+    str.each_line do |i|
+      next if i .include?('Calling core with action:') == false
```
Comment (labs-ruby) [inline HW_03/Boris_Tsarikov/boris_tsarikov_hw_03_t_03.rb]: line

File: HW_03/Boris_Tsarikov/boris_tsarikov_hw_03_array_to_hash_to_array.txt (hunk 1 above)
Comment (B0RlS) [reply HW_03/Boris_Tsarikov/boris_tsarikov_hw_03_array_to_hash_to_array.txt]: > don't understand, cold you please explain why are we need method doing nothing?

The task was the following: from an array to a hash, and from a hash to an array. Everything is working.
If we want to get a hash from an array, then just

script line:
Hash[(0...arr.size).zip arr].values.flatten
* Invisible transforms:
  {0=>1, 1=>3, 2=>6, 3=>8} - array to hash
*
Output result:
  [1, 3, 6, 8]

File: HW_03/Boris_Tsarikov/boris_tsarikov_hw_03_t_01.rb (hunk 2 above)
Comment (B0RlS) [reply HW_03/Boris_Tsarikov/boris_tsarikov_hw_03_t_01.rb]: > line instead i - more understandable

Ok, next time I will give more logical names to variables.

**Pull request merged**